Zend/zend_ast.c

```diff
@@ -1062,6 +1062,13 @@ static ZEND_COLD int zend_ast_valid_var_name(const char *s, size_t len)
 	return 1;
 }
 
+static ZEND_COLD int zend_ast_var_needs_braces(char ch)
+{
+	if (ch != '[' && !zend_ast_valid_var_char(ch))
```

Comment:
Maybe `if` is not necessary here:
```suggestion
	return ch != '[' && !zend_ast_valid_var_char(ch);
```